### ISSUE TITLE
Place to store filenames on attachments

### ISSUE THREAD

**Issue by lindafengsf** [IMSGlobal/caliper-central#207]
need to be able to represent filenames when emitting events.
Propose adding a storageName property to the digitalResource entity.

**Comment by ViktorHaag** [comment]
Thought a bit about this:
- we could add a storageName property on Entity and/or Event, so that all Caliper things could carry a storage name for a location in persistent storage
- why would this not just be the @id IRI of the thing, though?
- if we do want to have a separate storage name from the @id IRI, would the storageName also not have to be an IRI (with a `file` scheme)?

**Comment by arwhyte** [comment]
> * why would this not just be the @id IRI of the thing, though?

Yes, I was thinking this as well.  Possibly duplicative and unnecessary.

**Comment by ViktorHaag** [comment]
On further thought, I think this _is_ potentially useful. The underlying storage name might well be different to some particle of the URL and different to the "name" property of the thing. I think this does potentially make sense.

I'm also fine with limiting this for now to DigitalResource (for simplicity). This would be a v1p2 core change to digital resources.

**Comment by ViktorHaag** [comment]
It would probably be helpful for this one if we could have a use case recorded here in the issue?

**Comment by arwhyte** [comment]
Caliper Technical Working Group agreed to name this property `storageName`.  It is restricted to `DigitalResource` and subtypes only.
